mavros_extras/src/plugins/debug_value.cpp

```diff
@@ -0,0 +1,268 @@
+/**
+ * @brief Debug messages plugin
+ * @file debug_value.cpp
+ * @author Nuno Marques <n.marques21@hotmail.com>
+ *
+ * @addtogroup plugin
+ * @{
+ */
+/*
+ * Copyright 2017 Nuno Marques.
+ *
+ * This file is part of the mavros package and subject to the license terms
+ * in the top-level LICENSE file of the mavros repository.
+ * https://github.com/mavlink/mavros/tree/master/LICENSE.md
+ */
+
+#include <mavros/mavros_plugin.h>
+
+#include <mavros_msgs/DebugValue.h>
+
+namespace mavros {
+namespace extra_plugins {
+//! @brief Plugin for Debug msgs from MAVLink API
+class DebugValuePlugin : public plugin::PluginBase {
+public:
+	DebugValuePlugin() : PluginBase(),
+		debug_nh("~debug"),
+		publish_debug(false)
+	{ }
+
+	void initialize(UAS &uas_)
+	{
+		PluginBase::initialize(uas_);
+
+		// subscribers
+		debug_sub = debug_nh.subscribe("send", 10, &DebugValuePlugin::debug_cb, this);
+
+		debug_nh.param("publish_debug", publish_debug, false);
+
+		// publishers (optional)
+		if (publish_debug) {
+			debug_pub = debug_nh.advertise<mavros_msgs::DebugValue>("debug", 10);
+			/** @todo create a pub config based on the naming of the debug msgs,
+			 * using a map_dict as in distance_sensor plugin
+			 */
+			debug_vector_pub = debug_nh.advertise<mavros_msgs::DebugValue>("debug_vector", 10);
+			named_value_float_pub = debug_nh.advertise<mavros_msgs::DebugValue>("named_value_float", 10);
+			named_value_int_pub = debug_nh.advertise<mavros_msgs::DebugValue>("named_value_int", 10);
```

Comment:
I'm not sure that this really needed. I think that one common publisher is enough.

Reply:
if you publish msgs of different types, it is easier to track and to listen to them in different topics instead of one. I prefer this approach